### ISSUE TITLE
fix: mason update command not working(mason_cli)

### ISSUE THREAD

**Issue by tyypgzl** [felangel/mason#547]
**Description**
mason update command not working for me.
- i check my mason version.
```shell
➜  ~ mason --version
0.1.0-dev.35

Update available! 0.1.0-dev.35 → 0.1.0-dev.37
Changelog: https://github.com/felangel/mason/releases/tag/mason_cli-v0.1.0-dev.37
Run mason update to update
```
- I run mason update command.

```shell
➜  ~ mason update
✓ Checked for updates (0.4s)
✓ Updated to 0.1.0-dev.37 (1.8s)
➜  ~ 
```
- After the update, I check the mason version again.
```shell
➜  ~ mason --version
0.1.0-dev.35

Update available! 0.1.0-dev.35 → 0.1.0-dev.37
Changelog: https://github.com/felangel/mason/releases/tag/mason_cli-v0.1.0-dev.37
Run mason update to update
```
- still mason version 0.1.0-dev.35.

**Comment by felangel** [comment]
Hi @tyypgzl 👋
Thanks for opening an issue!

What version of Dart do you have installed? (`dart --version`)

I’m guessing you aren’t on >=2.16 so you need to upgrade dart first. Let me know if that helps 👍

**Comment by felangel** [comment]
Closing for now since there aren't any actionable next steps. Feel free to provide more info and/or a reproduction sample and I'm happy to take a closer look 👍

**Comment by tyypgzl** [comment]
Current dart sdk version

➜  ~ fvm dart --version
Dart SDK version: 2.18.2 (stable) (Tue Sep 27 13:24:11 2022 +0200) on "macos_arm64"

But i am not sure what dart version is when i get this error.

**Comment by felangel** [comment]
> Current dart sdk version
> 
> ➜ ~ fvm dart --version Dart SDK version: 2.18.2 (stable) (Tue Sep 27 13:24:11 2022 +0200) on "macos_arm64"
> 
> But i am not sure what dart version is when i get this error.

mason doesnt use fvm by default so you most likely have an older version of dart installed. You can check by running `dart --version` which likely needs to be updated or you need to configure dart to use the fvm installation

**Comment by tyypgzl** [comment]
current dart version
```
➜  ~ dart --version
Dart SDK version: 2.18.2 (stable) (Tue Sep 27 13:24:11 2022 +0200) on "macos_arm64"
```

**Comment by felangel** [comment]
> current dart version
> 
> ```
> ➜  ~ dart --version
> Dart SDK version: 2.18.2 (stable) (Tue Sep 27 13:24:11 2022 +0200) on "macos_arm64"
> ```

Did you install via hombrew or pub?

**Comment by tyypgzl** [comment]
via pub

**Comment by felangel** [comment]
> via pub

That should be resolved by #729. I'm planning to release a new version shortly but in the meantime the most likely cause of the issue is that you need to upgrade to the latest stable version of Dart (2.19) and try again. Let me know if that helps 👍
